### PR TITLE
Cr 1140 authentication timeout

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Change so that a user session timeout in RHUI always results in a timeout page, rather than an error reload of the UAC authentication page

# What has changed
Change to the HTTPForbidden resolution to render the standalone timeout page, rather than rendering the UAC authentication page
Change to remove flash message update for timeout
Restructure of direct access unit tests to new methods, and merging of existing tests to three super tests
No RHCucumber updates required

# How to test?
In RHUI, UAC authenticate, then allow session timeout or remove browser session cookie on various pages in the journey (before reaching EQ) and confirm that the dedicated timeout page is rendered, and that when reaching the UAC authentication page it does not display any in page error messages

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1140

### CODE REVIEW

**Review by PeterBochel — APPROVED** [review]
Approving with 1 minor comment

File: app/security.py [hunk 1]
```diff
@@ -83,7 +83,6 @@ async def check_permission(request):
                     url=request.rel_url.human_repr(),
                     client_ip=request['client_ip'])
     except KeyError:
-        flash(request, VALIDATION_FAILURE_MSG)
```
Comment (MartinHumphrey) [inline app/security.py:86]: Still import for _flash_ and _VALIDATION_FAILURE_MSG_ but are now no longer used. Fails flake8 rules.

**Review by MartinHumphrey — CHANGES_REQUESTED** [review]
Flake8 failure on unused imports needs fixing. Observation on different URL behaviour for different session timeout concerns depends on requirement, but I believe would be better if consistent?

**Review by MartinHumphrey — APPROVED** [review]
Approved, improvement and simplification of the code.
HTTPStatus 403 for HTTPForbidden errors where we are expecting a UAC to have been entered obviously OK but not sure about this HTTPStatus response for SessionTimeout errors where we do not have the required data from previous responses in the session. Not sure HTTP has an appropriate alternative status however? Maybe a question to raise through Phil?

**Pull request merged**